pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -75,6 +75,14 @@ static dtls_handler_t _dtls_handler = {
 #endif /* CONFIG_DTLS_ECC */
 };
 
+#ifdef CONFIG_DTLS_ECC
+/**
+ * @brief Array of ECDSA keys to convert from credman credentials and send in
+ *        the callback.
+ */
+static dtls_ecdsa_key_t _ecdsa_keys[CONFIG_DTLS_CREDENTIALS_MAX];
```

Comment:
If the corresponding credentials are removed with `sock_dtls_remove_credential`, will it also be removed from `_ecdsa_keys`?

Reply:
Currently not. It should actually not interfere with an ongoing session. Now that you mention this, probably `sock_dtls_session_destroy` should mark as free the previously used structure. Does tinyDTLS expose a pointer to the provided structure? If that's not the case, then probably that state should be kept externally (session <-> ecdsa struct).

Reply:
In d054da2597a45cb3acc4610d80a6df80293738a7 I modified the logic to keep a pointer to the session during the handshake. I did not find any documentation regarding when the key can be freed, currently this is done when the handshake is complete. Maybe we should check what happens on reconnections.